gpu_support/cuda_utils/install_cuda.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+install_cuda_version=$1
+cuda_install_dir=$2
+
+# TODO: Can we do a trimmed install?
+# Only install CUDA if specified version is not found.
+# This is only relevant for users, the shipped CUDA installation will
+# always be in versions instead of host_injections and have symlinks pointing
+# to host_injections for everything we're not allowed to ship
+if [ -f ${cuda_install_dir}/software/CUDA/${install_cuda_version}/EULA.txt ]; then
```

Comment:
The if/else is still good, except we should be checking under the `host_injections` path. This will allow us to skip any check on available space etc.

Reply:
You should construct `cuda_install_dir` rather than take it as an argument

Reply:
Also, we should allow for a forced installation to override this check

Reply:
It prefer the we ship the EULA text so I think we should check for an expectect broken symlink here:
```
if [[ -L "${cuda_install_dir}/software/CUDA/bin/nvcc" && -e "${cuda_install_dir}/software/CUDA/bin/nvcc" ]]; then
```